go-controller/pkg/metrics/ovn.go

```diff
@@ -279,6 +283,121 @@ func ovnControllerConfigurationMetricsUpdater() {
 	}
 }
 
+func ovnControllerServiceSubnetMetricsPublisher() {
+	var cidrV4 = ""
+	var cidrV6 = ""
+	var v4Size = 0
+	var v6Size = 0
+	for _, cidr := range config.Kubernetes.ServiceCIDRs {
+		ones, bits := cidr.Mask.Size()
+		size := bits - ones
+		if utilnet.IsIPv6CIDR(cidr) {
+			cidrV6 = cidr.String()
+			v6Size = size
+		} else {
+			cidrV4 = cidr.String()
+			v4Size = size
+		}
+	}
+
+	var metricServiceSubnet = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: MetricOvnkubeNamespace,
+		Subsystem: MetricOvnkubeSubsystemMaster,
+		Name:      "service_subnet",
+		Help: "A metric with a constant '1' value labeled by subnet that " +
+			"specifies the subnet used for service. subnet_v4 and subnet_v6 are " +
+			"used as labels for IPv4 and IPv6 subnets respectively. If any of the subnets " +
+			"is not configured, corresponding label will have a blank value.",
+	},
+		[]string{
+			"subnet_v4",
+			"subnet_v6",
+		},
+	)
+	ovnRegistry.MustRegister(metricServiceSubnet)
+	metricServiceSubnet.WithLabelValues(cidrV4, cidrV6).Set(1)
+
+	ovnRegistry.MustRegister(prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			Namespace: MetricOvnkubeNamespace,
+			Subsystem: MetricOvnkubeSubsystemMaster,
+			Name:      "service_subnet_size_ipv4",
+			Help:      "Size of IPv4 service subnet size. 0 for the value means IPv4 subnet is not configured.",
+		},
+		func() float64 {
+			return float64(v4Size)
+		}))
+
+	ovnRegistry.MustRegister(prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			Namespace: MetricOvnkubeNamespace,
+			Subsystem: MetricOvnkubeSubsystemMaster,
+			Name:      "service_subnet_size_ipv6",
+			Help:      "Size of IPv6 service subnet size. 0 for the value means IPv6 subnet is not configured.",
+		},
+		func() float64 {
+			return float64(v6Size)
+		}))
+}
+
+func ovnControllerClusterSubnetMetricsPublisher() {
+	var cidrV4 = ""
+	var cidrV6 = ""
+	var v4Size = 0
+	var v6Size = 0
+
+	for _, subnet := range config.Default.ClusterSubnets {
```

Comment:
Feel like some of this could be refactored into a generic method used for both `ClusterSubnets` and `ServiceCIDRs` and avoid some code duplication.

Reply:
done